examples/prom/grad_div_global_rom.cpp

```diff
@@ -0,0 +1,569 @@
+                     || (!offline && !merge && online);
+        MFEM_VERIFY(check, "only one of offline, merge, or online must be true!");
+    }
+
+    // 3. Enable hardware devices such as GPUs, and programming models such as
+    //    CUDA, OCCA, RAJA and OpenMP based on command line options.
+    Device device(device_config);
+    if (myid == 0) {
+        device.Print();
+    }
+
+    // 4. Read the (serial) mesh from the given mesh file on all processors.  We
+    //    can handle triangular, quadrilateral, tetrahedral, hexahedral, surface
+    //    and volume meshes with the same code.
+    Mesh mesh(mesh_file, 1, 1);
+    int dim = mesh.Dimension();
+    int sdim = mesh.SpaceDimension();
+
+    // 5. Refine the serial mesh on all processors to increase the resolution. In
+    //    this example we do 'ref_levels' of uniform refinement. We choose
+    //    'ref_levels' to be the largest number that gives a final mesh with no
+    //    more than 1,000 elements.
+    {
+        int ref_levels =
+            (int)floor(log(1000./mesh.GetNE())/log(2.)/dim);
+        for (int l = 0; l < ref_levels; l++)
+        {
+            mesh.UniformRefinement();
+        }
+    }
+
+    // 6. Define a parallel mesh by a partitioning of the serial mesh. Refine
+    //    this mesh further in parallel to increase the resolution. Once the
+    //    parallel mesh is defined, the serial mesh can be deleted.
+    ParMesh pmesh(MPI_COMM_WORLD, mesh);
+    mesh.Clear();
+    {
+        int par_ref_levels = 2;
+        for (int l = 0; l < par_ref_levels; l++)
+        {
+            pmesh.UniformRefinement();
+        }
+    }
+
+    // 7. Define a parallel finite element space on the parallel mesh. Here we
+    //    use the Raviart-Thomas finite elements of the specified order.
+    FiniteElementCollection *fec = new RT_FECollection(order-1, dim);
+    ParFiniteElementSpace fespace(&pmesh, fec);
+    HYPRE_BigInt size = fespace.GlobalTrueVSize();
+    if (myid == 0)
+    {
+        cout << "Number of finite element unknowns: " << size << endl;
+    }
+
+    // 8. Determine the list of true (i.e. parallel conforming) essential
+    //    boundary dofs. In this example, the boundary conditions are defined
+    //    by marking all the boundary attributes from the mesh as essential
+    //    (Dirichlet) and converting them to a list of true dofs.
+    Array<int> ess_tdof_list;
+    if (pmesh.bdr_attributes.Size())
+    {
+        Array<int> ess_bdr(pmesh.bdr_attributes.Max());
+        ess_bdr = set_bc ? 1 : 0;
+        fespace.GetEssentialTrueDofs(ess_bdr, ess_tdof_list);
+    }
+
+    // 9. Initiate ROM related variables
+    int max_num_snapshots = 100;
+    bool update_right_SV = false;
+    bool isIncremental = false;
+    const std::string basisName = "basis";
+    const std::string basisFileName = basisName + std::to_string(id);
+    const CAROM::Matrix* spatialbasis;
```

Comment:
`spatialbasis` could be declared only in the `online` case, as well as some other variables such as `numRowRB`, etc.